From 4780a01804dfa5215823d19370384264762a6b0e Mon Sep 17 00:00:00 2001
From: nikhilad <nikhilad@ami.com>
Date: Fri, 18 Apr 2025 08:36:09 +0530
Subject: [PATCH] Generating a signal for power operations

For Power Off, generating a signal for RequestedPowerTransition
For Power cycle and reset, for consecutive operation generating
a signal or RequestedHostTransition

Signed-off-by: nikhilad <nikhilad@ami.com>
---
 src/power_control.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index 79ae2ca..823e0ec 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -260,6 +260,13 @@ static void beep(const uint8_t& beepPriority)
         "xyz.openbmc_project.BeepCode", "Beep", uint8_t(beepPriority));
 }
 
+static void
+    sendSignal(const std::shared_ptr<sdbusplus::asio::dbus_interface> iface,
+               const std::string property)
+{
+    iface->signal_property(property);
+}
+
 enum class OperatingSystemStateStage
 {
     Inactive,
@@ -3325,6 +3332,10 @@ int main(int argc, char* argv[])
                     return 0;
                 }
             }
+
+            if (resp == requested)
+                sendSignal(hostIface, "RequestedHostTransition");
+
             resp = requested;
             return 1;
         });
@@ -3497,6 +3508,10 @@ int main(int argc, char* argv[])
                 throw std::invalid_argument("Unrecognized Transition Request");
                 return 0;
             }
+
+            if (resp == requested)
+                sendSignal(chassisIface, "RequestedPowerTransition");
+
             resp = requested;
             return 1;
         });
-- 
2.25.1

